File: web/dart/versions/version-2.dart
part of NetTango;

class Version2 {

  static void update(js.JsObject json) {
    VersionUtils.updateBlocks3(json, updateBlockSelectAttributes, updateBlockSelectAttributes);
  }

  static void updateBlockSelectAttributes(js.JsObject b) {
    VersionUtils.updateBlockAttributes(b, objectifySelectAttributes);
  }

  static void objectifySelectAttribute(js.JsObject attribute) {
    if (!attribute.hasProperty("values") || attribute["values"] is! js.JsArray) {
      return;
    }
    attribute["values"] = js.JsArray.from(attribute["values"].map( (v) { return js.JsObject.jsify({ "actual": v }); } ));
  }

  static void objectifySelectAttributes(js.JsArray attributes) {
    for (js.JsObject a in attributes.where( (f) => f.hasProperty("type") && f["type"] == "select" )) {
      objectifySelectAttribute(a);
    }
  }

}

File: src/versions/version-2.ts
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

class Version2 {

  static update(json: any): void {
    VersionUtils.updateBlocks3(json, Version2.updateBlockSelectAttributes, Version2.updateBlockSelectAttributes)
  }

  static updateBlockSelectAttributes(b: any): void {
    VersionUtils.updateBlockAttributes(b, Version2.objectifySelectAttributes)
  }

  static objectifySelectAttribute(attribute: any): void {
    if (!attribute.hasOwnProperty("values") || !Array.isArray(attribute["values"])) {
      return
    }
    attribute["values"] = attribute["values"].map( (v) => { return { "actual": v } } )
  }

  static objectifySelectAttributes(attributes: any[]): void {
    for (var a of attributes.filter( (f) => f.hasOwnProperty("type") && f["type"] === "select" )) {
      Version2.objectifySelectAttribute(a)
    }
  }

}
